src/ophys_etl/types/_roi.py

```diff
@@ -286,3 +286,114 @@ def contour_mask(self) -> np.ndarray:
         if self._contour_mask is None:
             self._construct_contour_mask()
         return np.copy(self._contour_mask)
+
+    def get_bounding_box_cutout(self, image: np.ndarray) -> np.ndarray:
+        """Return a cutout from an image that matches the roi bounding box.
+        """
+        return image[self._y0:self._y0 + self._height,
+                     self._x0:self._x0 + self._width]
+
+    def get_centered_cutout(self,
+                            image: np.ndarray,
+                            height: int,
+                            width: int) -> np.ndarray:
+        """Get a cutout of arbitrary size centered on the bounding box
+        centroid.
+
+        Pad cutout with zeros if requested cutout extends outside of image.
+
+        Parameters
+        ----------
+        image : numpy.ndarray, (N, M)
+            Image to create cutout/thumbnail from.
+        height : int
+            Height(y) of output cutout image.
+        width : int
+            Width(x) of output cutout image.
+
+        Returns
+        -------
+        cutout : numpy.ndarray
+            Cutout of requested size centered on the bounding box center.
+        """
+        # Find the indices of the desired cutout in the image.
+        row_indices = self._get_cutout_indices(self.bounding_box_center_y,
+                                               image.shape[-2],
```

Comment:
Is there a reason you didn't use `image.shape[0]` here? I think that might be more readable. Same comment for `image.shape[-1]` below.

Reply:
I uses these indexes as I had a half backed idea of being able to use this movies as well as images. Didn't fully implement it so I can change the indices.